language/stdlib/modules/AccountLimits.move

```diff
@@ -1,261 +1,264 @@
+        let limits_def = borrow_global_mut<LimitsDefinition<CoinType>>(limit_address);
+        if (new_max_inflow > 0) { limits_def.max_inflow = new_max_inflow };
+        if (new_max_outflow > 0) { limits_def.max_outflow = new_max_outflow };
+        if (new_max_holding_balance > 0) { limits_def.max_holding = new_max_holding_balance };
     }
 
     ///////////////////////////////////////////////////////////////////////////
     // Internal utiility functions
     ///////////////////////////////////////////////////////////////////////////
 
-    // If the time window starting at `window.window_start` and lasting for
-    // `limits_definition.time_period` has elapsed, resets the window and
-    // the inflow and outflow records. Additionally the new
-    // `tracked_balance` is computed at this time as well.
-    fun reset_window(window: &mut Window, limits_definition: &LimitsDefinition) {
+    /// If the time window starting at `window.window_start` and lasting for
+    /// `limits_definition.time_period` has elapsed, resets the window and
+    /// the inflow and outflow records.
+    fun reset_window<CoinType>(window: &mut Window<CoinType>, limits_definition: &LimitsDefinition<CoinType>) {
         let current_time = LibraTimestamp::now_microseconds();
         if (current_time > window.window_start + limits_definition.time_period) {
             window.window_start = current_time;
-            window.window_total_flow = 0;
+            window.window_inflow = 0;
+            window.window_outflow = 0;
         }
     }
 
-    // Verify that the receiving account tracked by the `receiving` window
-    // can receive `amount` funds without violating requirements
-    // specified the `limits_definition` passed in.
+    /// Verify that the receiving account tracked by the `receiving` window
+    /// can receive `amount` funds without violating requirements
+    /// specified the `limits_definition` passed in.
     fun can_receive<CoinType>(
         amount: u64,
-        receiving: &mut Window,
+        receiving: &mut Window<CoinType>,
     ): bool acquires LimitsDefinition {
-        let limits_definition = borrow_global_mut<LimitsDefinition>(receiving.limits_definition);
-        // If the limits ares unrestricted then no more work needs to be done
-        if (is_unrestricted(limits_definition)) return true;
+        let limits_definition = borrow_global_mut<LimitsDefinition<CoinType>>(receiving.limit_address);
+        // If the limits are unrestricted then only track the balance. We track
+        // the balance even if the limits are "unlimited" since otherwise there
+        // is no way to know the combined balance of an entity on-chain, this
+        // would then preclude setting balance holding limits on an entity at a future date.
+        if (is_unrestricted(limits_definition)) {
+            /*receiving.tracked_balance = receiving.tracked_balance + amount;*/
```

Comment:
This and the other line in `can_withdraw` are causing a serious perf regression; on the order of about 230TPS. Not sure what we should do about it... any changing of the parent VASP blob is going to cause this issue.

Reply:
How did you get a regression breakdown by method?